scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+-----------------------------------
+-- PET: Avatar
+-----------------------------------
+require('scripts/globals/summon')
+-----------------------------------
+
+xi = xi or {}
+xi.pets = xi.pets or {}
+xi.pets.avatar = {}
+
+local buffModeVar          = 'AVATAR_BUFF_MODE_OFF'
+local lastCastTimeVar      = 'AVATAR_LAST_CASTINGTIME'
+local lastCastTimeStampVar = 'AVATAR_LAST_CAST_TIMESTAMP'
+local playerListenerVar    = 'SMN_SPIRIT_CAST_DELAY'
+
+-- will determine, in seconds, the spirit's casting cooldown
+-- Called every tick for player to adjust magic casting delay in real-time
+local setMagicCastCooldown = function(pet)
+    local castingCooldown = 45
+    local master = pet:getMaster()
```

Comment:
There's a few sections in here where you nest the entire function inside `if master`, you can reduce the nesting by exiting early:

```lua
if not master then
    return
end
```

Reply:
Done